.github/workflows/on-pullrequest.yml

```diff
@@ -1,10 +1,15 @@
-name: Test
+name: On Pull Request
 
 on:
   push:
-    branches: [main]
+    branches:
```

Comment:
For Meadowlark, we run the tests once the PR is open and after it's merged. This is contradictory with the name, but I consider this is a good practice. If required, we can remove this trigger to leave only as "On Pull Request".

Reply:
I was thinking of an scenario where a PR is opened, and then stays open for a couple of weeks while there are other PRs getting merged but since we don't allow automatic merges I don't think this scenario is feasible.

I don't think this adds any value unless scheduled or something similar, given that even if something changes on main, the workflow will not be executed unless there's a commit, which is not allowed. 

Therefore, I don't think we need this push on main

Reply:
In the daily standup we decided that there is a very small, but real, risk of something going wrong during a merge, and therefore it will be better to go ahead and test on merge.